R/Artifact.R

```diff
@@ -46,6 +46,35 @@ ArtifactRecord <- R6::R6Class( # nolint object_name_linter
       }
     },
     #' @description
+    #' Return a backed data object. Currently only supports TileDB-SOMA
+    #' artifacts.
+    #'
+    #' @return A [tiledbsoma::SOMACollection] or [tiledbsoma::SOMAExperiment]
+    #' object
+    open = function() {
+      is_tiledbsoma <- private$get_value("suffix") == ".tiledbsoma" ||
+        private$get_value("_accessor") == "tiledbsoma"
+
+      if (!is_tiledbsoma) {
+        cli::cli_abort(
+          "The {.code open} method is only supported for TileDB-SOMA artifacts"
+        )
+      }
+
+      check_requires(
+        "Opening TileDB-SOMA artifacts", "tiledbsoma",
+        extra_repos = "https://chanzuckerberg.r-universe.dev"
+      )
+
+      artifact_uri <- paste0(
```

Comment:
The problem is that it won't work for s3 paths i think. At least the correct region should be provided, we have additional setup here https://github.com/laminlabs/lamindb/blob/39e0f529a41d9cbc475e52fe3aa0b8095af21494/lamindb/core/storage/_tiledbsoma.py#L33 but not sure what is available in R.